source/common/http/header_map_impl.h

```diff
@@ -48,6 +48,11 @@ class HeaderMapImpl : public HeaderMap {
   void addViaMove(HeaderString&& key, HeaderString&& value);
 
   /**
+   * Set a header via full move.
+   */
+  void setViaMove(HeaderString&& key, HeaderString&& value);
```

Comment:
Does this need to be public?

Reply:
No, in which case it doesn't even need to exist.